app/jobs/maintenance_tasks/task_job.rb

```diff
@@ -97,8 +97,18 @@ def on_shutdown
       @ticker.persist
     end
 
+    def reenqueue_iteration_job
+      if defined?(@reenqueue_iteration_job) && @reenqueue_iteration_job
+        super
+      end
+    end
+
     def after_perform
       @run.save!
+      if @run.interrupted?
+        @reenqueue_iteration_job = true
+        reenqueue_iteration_job
```

Comment:
If we're keeping the logic around `@run.interrupted?`, we could simplify by not using an instance variable, just a parameter to `reenqueue_iteration_job`.